### PR TITLE
Works on student correction form.

### PR DESCRIPTION
**Pull request by BhagyashreeBhoyar**

### CODE REVIEW

File: src/components/formComponents/SelectListInputField.jsx [hunk 1]
```diff
@@ -18,17 +18,17 @@ class SelectListInputField extends Component {
     return this.props.options.map((iterator, index) => {
       return <option value={iterator.value} key={index}>{iterator.text}</option>
     });
-  };
+  }
 
   populateValue(value) {
-    if(value === '' || value === null || value === undefined) {
+    if (value === '' || value === null || value === undefined) {
```
Comment (mohitsaladiya) [inline src/components/formComponents/SelectListInputField.jsx]:
```suggestion
    if (!value) {
```

File: src/components/StudentCredentialPage.jsx [hunk 2]
```diff
@@ -121,6 +123,7 @@ class StudentCredentialPage extends Component {
           </div>
         );
       } else if (this.props.studentData && this.props.isFetched) {
+        this.props.setUserTypeAction('student');
```
Comment (mohitsaladiya) [inline src/components/StudentCredentialPage.jsx]: create a constant USER_TYPES = {
STUDENT: 'student',
ADMIN: 'admin'
}
```suggestion
        this.props.setUserTypeAction(USER_TYPES.STUDENT);
```

File: src/components/StudentInformationGrid.jsx [hunk 3]
```diff
@@ -311,6 +312,7 @@ class StudentInformationGrid extends Component {
       this.props.fetchStudentData(String(rowData.studentId), adminPassword);
       this.props.setStudentDataAction(this.props.studentData);
       this.props.updateStudentByAdminAction(String(rowData.studentId), adminPassword);
+      this.props.setUserTypeAction('admin');
```
Comment (mohitsaladiya) [inline src/components/StudentInformationGrid.jsx]: Use constant here.

File: src/components/StudentRegistrationCorrectionForm.jsx [hunk 4]
```diff
@@ -104,6 +118,154 @@ class StudentRegistrationCorrectionForm extends Component {
...
+      this.checkError({ email: '', motherMobile: '' });
+    }
+  }
+
+  /**
+   * renderAdminEditableFields render admin editable field which will edit by only admin.
+   * @return {ReactComponent}
+   */
+  renderAdminEditableFields = () => {
+    if (this.props.pageUser === 'admin') {
```
Comment (mohitsaladiya) [inline src/components/StudentRegistrationCorrectionForm.jsx]: Use constant here.

File: src/components/StudentRegistrationCorrectionForm.jsx [hunk 5]
```diff
@@ -112,42 +274,77 @@ class StudentRegistrationCorrectionForm extends Component {
       errorMessage: errorMessageObject,
     });
   }
+
+  /**
+   * renderLavelField render Level field according to user type.
+   * @return {Reactcomponent}
+   */
+  renderLevelField = () => {
+    if (this.props.pageUser === 'admin') {
```
Comment (mohitsaladiya) [inline src/components/StudentRegistrationCorrectionForm.jsx]: Use constant.

File: src/components/StudentRegistrationCorrectionForm.jsx [hunk 6]
```diff
@@ -112,42 +274,77 @@ class StudentRegistrationCorrectionForm extends Component {
...
+        isRequired
+      />
+    );
+  };
+  /**
+   * renderBackButton method render back button according to user type
+   * @return {ReactComponent}
+   */
+  renderBackButton = () => {
+    if (this.props.pageUser === 'admin') {
```
Comment (mohitsaladiya) [inline src/components/StudentRegistrationCorrectionForm.jsx]: Use constant.

File: src/components/StudentRegistrationCorrectionForm.jsx [hunk 7]
```diff
@@ -112,42 +274,77 @@ class StudentRegistrationCorrectionForm extends Component {
...
+      return (
+        <LinkButton
+          buttonText={goBackBtnText}
+          linkPath={this.props.context.previousLocation}
+        />
+      );
+    } else if (this.props.pageUser === 'url') {
```
Comment (mohitsaladiya) [inline src/components/StudentRegistrationCorrectionForm.jsx]: URL can't be a user. Please use a specific name. 
And import it from constant here.

File: src/components/formComponents/SelectListInputField.jsx (hunk 1 above)
Comment (BhagyashreeBhoyar) [reply src/components/formComponents/SelectListInputField.jsx]: Done.

File: src/components/StudentCredentialPage.jsx (hunk 2 above)
Comment (BhagyashreeBhoyar) [reply src/components/StudentCredentialPage.jsx]: Done.

File: src/components/StudentInformationGrid.jsx (hunk 3 above)
Comment (BhagyashreeBhoyar) [reply src/components/StudentInformationGrid.jsx]: Done.

File: src/components/StudentRegistrationCorrectionForm.jsx (hunk 4 above)
Comment (BhagyashreeBhoyar) [reply src/components/StudentRegistrationCorrectionForm.jsx]: Done.

File: src/components/StudentRegistrationCorrectionForm.jsx (hunk 5 above)
Comment (BhagyashreeBhoyar) [reply src/components/StudentRegistrationCorrectionForm.jsx]: Done.

File: src/components/StudentRegistrationCorrectionForm.jsx (hunk 6 above)
Comment (BhagyashreeBhoyar) [reply src/components/StudentRegistrationCorrectionForm.jsx]: Done.

File: src/components/StudentRegistrationCorrectionForm.jsx (hunk 7 above)
Comment (BhagyashreeBhoyar) [reply src/components/StudentRegistrationCorrectionForm.jsx]: Done.

**Pull request merged**